Creating question form

StinaSoile opened the pull request:
opens next pages when questions are answered correctly. Accepts only one specific form of right answer (32, no thirty two, etc). Button flashes green if right answer, and red if wrong answer.
Try with bat-comic, the other comic does not yet have questions at all.

StinaSoile committed:
renamed components
created new component MathQuestions.tsx

StinaSoile committed:
komponentit jaettu kansioihin

StinaSoile committed:
Did same changes as in connecting backend -branch:
- fixed the bad code in comicGallery
- changed NODE_ENV to VITE_BACKEND
- added unique keys to list elements in app.tsx
- removed one console.log from comicBook.tsx

StinaSoile committed:
renaming css-style list-element-for-comicgallery

StinaSoile committed:
oikea vastaus avaa seuraavat sivut

StinaSoile committed:
fix:
korjattu vastauskentän tyhjentämisestä aiheutunut tilanne, jossa backend ei aina osannut tulkita vastausta stringina.

StinaSoile committed:
animations for wrong and right answers

StinaSoile committed:
changed the way comic pages shrink in the view with all comic pages:
now using media query.

StinaSoile committed:
trims answers before sending to backend

sepi4 commented on src/types.ts:
```diff
@@ -6,13 +6,13 @@ export interface Page {
-  pictureName: string;
-  questionList?: QuestionWithNoAnswer[];
-}
+// export interface PageWithNoAnswer {
```
remove commented code

sepi4 commented on src/components/ComicBook/ComicBook.tsx:
```diff
@@ -3,23 +3,33 @@ import "./ComicPage.css";
-    setKey(undefined);
-    getPages(comicName, key);
+    getPages(comicName, stateKey);
+    console.log(stateKey);
```
remove console.log

sepi4 commented on src/components/ComicBook/ComicPage.css:
```diff
@@ -0,0 +1,170 @@
+  align-items: center;
+}
+
+/* .all-questions-container {
```
commets

sepi4 commented on src/components/ComicBook/MathQuestions.tsx:
```diff
@@ -0,0 +1,142 @@
+    if (comic[page].questionList) {
+      const l = comic[page].questionList.length;
+      for (let q = 0; q < l; q++) {
+        const question = comic[page].questionList[q];
```
do not use `for` loop with indexing, use `for of`, `.map` or `.forEach`. Use normal `for` when you need to use indexing.

sepi4 commented on src/components/ComicBook/MathQuestions.tsx:
```diff
@@ -0,0 +1,142 @@
+  stateKey: string | undefined;
+  setKey: React.Dispatch<React.SetStateAction<string | undefined>>;
+  comicName: string;
+  comic: Page[];
```
Are you sure you need to send comic and page number, and then index page from comic? Is it possible to give one Page as a prop. Page should have questions in it?

sepi4 commented on src/components/ComicBook/MathQuestions.tsx:
```diff
@@ -0,0 +1,142 @@
+    setButtonClasses("answer-button wrong-answer");
+    setTimeout(() => {
+      setButtonClasses("answer-button");
+    }, 500);
```
this is too short for me at least

StinaSoile replied:
Changed to 1 sec. Still too short?

sepi4 replied:
better

sepi4 commented on src/components/ComicBook/MathQuestions.tsx:
```diff
@@ -0,0 +1,142 @@
+  const handleAnswer = async (e: SyntheticEvent) => {
+    e.preventDefault();
+    if (
+      comic[page].questionList &&
```
Maybe this `if (...)` can be cleand? To own function 
```
if(thereAreAnswers(comic[page]) {
...
```

StinaSoile replied:
typescript says that questionList might be undefined, if I move these to a separate function. I did that, though. Now it just checks in two places if the same questionList exists: 
in that if(...), and in the function thereAreAnswers().

sepi4 commented:
try using conventional commits 
https://gist.github.com/qoomon/5dfcdf8eec66a051ecd85625518cfd13

StinaSoile committed:
refactor:
- change for loop to .map
-  change props so that MathQuestions are only given one page instead of all pages
- clear if-statement in handleAnswer to its own function
chore:
- remove comments
- remove console.log
- make button animation longer

sepi4 commented:
Always have short heading in commit messages. Add more text if need on different lines
![image](https://github.com/user-attachments/assets/a574a9be-e0a3-4ff5-a037-99ca0867c84b)

sepi4 commented:
You can try reword your commit messages and squash you commit (if you dare 👹).
